tests/setup-stress-dependencies.sh

```diff
@@ -101,23 +96,81 @@ if [ ! -e $libInstallDir ]; then
     mkdir -p $libInstallDir
 fi
 
+# Use uname to determine what the OS is.
+OSName=$(uname -s)
+case $OSName in
+    Linux)
+        __BuildOS=Linux
+        __HostOS=Linux
+        ;;
+
+    Darwin)
+        __BuildOS=OSX
+        __HostOS=OSX
+        ;;
+
+    FreeBSD)
+        __BuildOS=FreeBSD
+        __HostOS=FreeBSD
+        ;;
+
+    OpenBSD)
+        __BuildOS=OpenBSD
+        __HostOS=OpenBSD
+        ;;
+
+    NetBSD)
+        __BuildOS=NetBSD
+        __HostOS=NetBSD
+        ;;
+
+    SunOS)
+        __BuildOS=SunOS
+        __HostOS=SunOS
+        ;;
+
+    *)
+        echo "Unsupported OS $OSName detected, configuring as if for Linux"
+        __BuildOS=Linux
+        __HostOS=Linux
+        ;;
+esac
+
+isPortable=0
+
+source "${scriptDir}"/../init-distro-rid.sh
+initDistroRidGlobal ${__BuildOS} x64 ${isPortable}
+
+# Hack, replace the rid to ubuntu.14.04 which has a valid non-portable
```

Comment:
Do we plan to eventually remove this hack?

Reply:
Yes